packages/kit/src/lib/HookSystem.ts

```diff
@@ -79,11 +123,21 @@ export class HookError<TError = Error | unknown> extends Error {
 	}
 }
 
+/**
+ * Next hook ID, incremented each time a hook is registered.
+ */
+let nextHookID = 0;
```

Comment:
Could we abstract ID generation to a function? That would let us change the implementation more easily if we decide to do so in the future.

```typescript
// Adapted from `@prismicio/richtext`

const generateID = (): string => {
	return (++generateID.state).toString();
};
generateID.state = 0;
```

Reply:
Goo call! Reused the function from `@prismicio/richtext` (`uuid()`)